src/wickrcrypto/src/buffer.c

```diff
@@ -30,15 +30,15 @@ static wickr_buffer_t *__wickr_buffer_create_empty(size_t len, void *(*alloc_fun
         return NULL;
     }
     
-    uint8_t *bytes = alloc_func(len + sizeof(wickr_buffer_t));
-    
+    uint8_t *bytes = alloc_func(sizeof(wickr_buffer_t));
+
     if (!bytes) {
         return NULL;
     }
     
     wickr_buffer_t *new_buffer = (wickr_buffer_t *)bytes;
-    
-    new_buffer->bytes = (uint8_t *)(new_buffer + 1);
+
+    new_buffer->bytes = alloc_func(len);
```

Comment:
@spompelio can you just do a check like above on line 36 to return null if the allocation of bytes fails?